src/wallet/rpcdump.cpp

```diff
@@ -151,27 +152,32 @@ UniValue importprivkey(const JSONRPCRequest& request)
         if (!pwalletMain->AddKeyPubKey(key, pubkey))
             throw JSONRPCError(RPC_WALLET_ERROR, "Error adding key to wallet");
 
+        if (fRescan)
+            attemptRescanFromHeight(request);
+    }
+
+    return NullUniValue;
+}
 
-        if (fRescan) {
-            CBlockIndex* pblockindex = chainActive.Genesis();
+void attemptRescanFromHeight(const JSONRPCRequest& request)
```

Comment:
Good idea to generalize this into a utility function! I have some thoughts about the structure proposed but am open to counter-arguments, see last line:

1. I'm not sure about validating the request inside the helper function, especially because it only looks at a single parameter in a hardcoded position (`3`) and think it would be more readable to do the validation inside the controller (`ImportAddress`) and make `attemptRescanForHeight(request)` into a more definite `rescanFromHeight(int)`? That would make the controller decide what to do with user input and the rescan utility function just do what is decided?
2. To reduce code duplication we could make the height validation generic (`RpcValidateChainHeight(int)`)?

Right now, the remaining `importmulti` would also get a rescan height at position 3 (I'd guess), so ultimately I think the question is: **are we minimizing the impact of this change, or minimizing the impact of any future work?**

Reply:
Good question. When you put it that way, if we ask *what code will we want to re-read in the future*, splitting this code into two helper functions is clearly better, because they can each have a single responsibility.

I don't mind doing a little extra work now to require a little less brainpower and research in the future.